src/sentry/tasks/commit_context.py

```diff
@@ -0,0 +1,157 @@
+from sentry.models.groupowner import GroupOwner, GroupOwnerType
+from sentry.ownership.grammar import get_source_code_path_from_stacktrace_path
+from sentry.tasks.base import instrumented_task
+from sentry.utils import metrics
+from sentry.utils.event_frames import munged_filename_and_frames
+from sentry.utils.locking import UnableToAcquireLock
+from sentry.utils.sdk import set_current_event_project
+
+PREFERRED_GROUP_OWNERS = 1
+PREFERRED_GROUP_OWNER_AGE = timedelta(days=7)
+
+logger = logging.getLogger("tasks.commit_context")
+
+
+@instrumented_task(
+    name="sentry.tasks.process_commit_context",
+    queue="group_owners.process_commit_context",
+    default_retry_delay=5,
+    max_retries=5,
+)
+def process_commit_context(
+    event_id, event_platform, event_frames, group_id, project_id, sdk_name=None, **kwargs
+):
+    lock = locks.get(
+        f"process-commit-context:{group_id}", duration=10, name="process_commit_context"
+    )
+    try:
+        with lock.acquire():
+            metrics.incr("sentry.tasks.process_commit_context.start")
+            set_current_event_project(project_id)
+
+            project = Project.objects.get_from_cache(id=project_id)
+            owners = GroupOwner.objects.filter(
+                group_id=group_id,
+                project=project,
+                organization_id=project.organization_id,
+                type=GroupOwnerType.SUSPECT_COMMIT.value,
+            )
+
+            # Delete old owners
+            to_be_deleted = owners.filter(
+                date_added__gte=timezone.now() - PREFERRED_GROUP_OWNER_AGE
+            )
+
+            if len(to_be_deleted):
+                for record in to_be_deleted:
+                    record.delete()
+
+            current_owners = owners.filter(
+                date_added__lte=timezone.now() - PREFERRED_GROUP_OWNER_AGE
+            ).order_by("-date_added")
+
+            if len(current_owners) >= PREFERRED_GROUP_OWNERS:
+                metrics.incr(
+                    "sentry.tasks.process_commit_context.aborted",
+                    tags={"detail": "maxed_owners_none_old"},
+                )
+                return
+
+            code_mappings = RepositoryProjectPathConfig.objects.filter(project=project)
+
+            frames = event_frames or []
+            munged = munged_filename_and_frames(event_platform, frames, "munged_filename", sdk_name)
+            if munged:
+                frames = munged[1]
+
+            # First frame in the stacktrace that is "in_app"
+            frame = next(filter(lambda frame: frame.get("in_app", False), frames), None)
+
+            if not frame:
+                metrics.incr(
+                    "sentry.tasks.process_commit_context.aborted",
+                    tags={"detail": "could_not_find_in_app_stacktrace_frame"},
+                )
+                return
+
+            commit_context = None
+            selected_code_mapping = None
+            for code_mapping in code_mappings:
```

Comment:
Instead of this for loop, I would have a function that returns the source path by trying which code mapping makes a change and matches a source file. That way you can get a verified source path rather than trying to get the commit context for each potential source path.

I would move such function to some module that has to do with code mappings or stack trace and name it something like `get_mapped_source_file_from_stack_trace`.